C error when assigning variable to const int_type ptr

### Describe the bug

Builder error:
```v
const x = 3

fn main() {
	_ := &x
}
```
Works:
```v
const x = int(3)

fn main() {
	_ := &x
}
```

### Expected Behavior

Both versions work.

### Current Behavior

```
/tmp/v_1000/const_ptr.13403185457188648864.tmp.c:12442: error: lvalue expected
builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

### Reproduction Steps

above

### Possible Solution

https://github.com/vlang/v/blob/815439a381b9f6ae1e39d373793c2407e7c4f224/vlib/v/gen/c/cgen.v#L5311-L5319

### Additional Information/Context

_No response_

### V version

0.4.0 815439a

### Environment details (OS name and version, etc.)

OS: linux, Pop!_OS 22.04 LTS
Processor: 16 cpus, 64bit, little endian, 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz

getwd: /home/turiiya/Dev/vlang/v
vexe: /home/turiiya/Dev/vlang/v/v
vexe mtime: 2023-08-16 05:27:47

vroot: OK, value: /home/turiiya/Dev/vlang/v
VMODULES: OK, value: /home/turiiya/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.34.1
Git vroot status: weekly.2023.33
.git/config present: true

CC version: cc (Ubuntu 12.3.0-1ubuntu1~22.04) 12.3.0
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

--- a/vlib/v/checker/assign.v
+++ b/vlib/v/checker/assign.v
@@ -264,15 +264,15 @@ fn (mut c Checker) assign_stmt(mut node ast.AssignStmt) {
 		node.left_types << left_type
 		match mut left {
 			ast.Ident {
-				if is_decl || left.kind == .blank_ident {
-					if left_type.is_ptr() && mut right is ast.PrefixExpr
-						&& right.right_type == ast.int_literal_type_idx {
-						if mut right.right is ast.Ident && right.right.obj is ast.ConstField {
-							c.error('cannot assign a pointer to a constant with an integer literal value',
-								right.right.pos)
-							c.add_error_detail('Specify the type for the constant value. Example:')
-							c.add_error_detail('         `const ${right.right.name.all_after_last('.')} = int(${(right.right.obj as ast.ConstField).expr})`')
-						}
+				if (is_decl || left.kind == .blank_ident) && left_type.is_ptr()
+					&& mut right is ast.PrefixExpr && right.right_type == ast.int_literal_type_idx {
+					if mut right.right is ast.Ident && right.right.obj is ast.ConstField {
+						const_name := right.right.name.all_after_last('.')
+						const_val := (right.right.obj as ast.ConstField).expr
+						c.error('cannot assign a pointer to a constant with an integer literal value',
+							right.right.pos)
+						c.add_error_detail('Specify the type for the constant value. Example:')
+						c.add_error_detail('         `const ${const_name} = int(${const_val})`')
 					}
 				}
 				if left.kind == .blank_ident {

--- a/vlib/v/checker/assign.v
+++ b/vlib/v/checker/assign.v
@@ -269,13 +269,12 @@ fn (mut c Checker) assign_stmt(mut node ast.AssignStmt) {
 					if mut right.right is ast.Ident && right.right.obj is ast.ConstField {
 						const_name := right.right.name.all_after_last('.')
 						const_val := (right.right.obj as ast.ConstField).expr
-						c.error('cannot assign a pointer to a constant with an integer literal value',
-							right.right.pos)
 						c.add_error_detail('Specify the type for the constant value. Example:')
 						c.add_error_detail('         `const ${const_name} = int(${const_val})`')
+						c.error('cannot assign a pointer to a constant with an integer literal value',
+							right.right.pos)
 					}
-				}
-				if left.kind == .blank_ident {
+				} else if left.kind == .blank_ident {
 					left_type = right_type
 					node.left_types[i] = right_type
 					if node.op !in [.assign, .decl_assign] {

--- a/vlib/v/checker/assign.v
+++ b/vlib/v/checker/assign.v
@@ -289,6 +289,15 @@ fn (mut c Checker) assign_stmt(mut node ast.AssignStmt) {
 							c.error('use of untyped nil in assignment (use `unsafe` | ${c.inside_unsafe})',
 								right.pos())
 						}
+						if mut right is ast.PrefixExpr && left_type.is_ptr()
+							&& right.right_type == ast.int_literal_type_idx {
+							if mut right.right is ast.Ident && right.right.obj is ast.ConstField {
+								c.error('cannot assign a pointer to a constant with an integer literal value',
+									right.right.pos)
+								c.add_error_detail('Specify the type for the constant value. Example:')
+								c.add_error_detail('         `const ${right.right.name.all_after_last('.')} = int(${(right.right.obj as ast.ConstField).expr})`')
+							}
+						}
 					}
 					mut ident_var_info := left.info as ast.IdentVar
 					if ident_var_info.share == .shared_t {

--- a/vlib/v/checker/assign.v
+++ b/vlib/v/checker/assign.v
@@ -264,6 +264,17 @@ fn (mut c Checker) assign_stmt(mut node ast.AssignStmt) {
 		node.left_types << left_type
 		match mut left {
 			ast.Ident {
+				if is_decl || left.kind == .blank_ident {
+					if left_type.is_ptr() && mut right is ast.PrefixExpr
+						&& right.right_type == ast.int_literal_type_idx {
+						if mut right.right is ast.Ident && right.right.obj is ast.ConstField {
+							c.error('cannot assign a pointer to a constant with an integer literal value',
+								right.right.pos)
+							c.add_error_detail('Specify the type for the constant value. Example:')
+							c.add_error_detail('         `const ${right.right.name.all_after_last('.')} = int(${(right.right.obj as ast.ConstField).expr})`')
+						}
+					}
+				}
 				if left.kind == .blank_ident {
 					left_type = right_type
 					node.left_types[i] = right_type
@@ -288,15 +299,6 @@ fn (mut c Checker) assign_stmt(mut node ast.AssignStmt) {
 							// }`
 							c.error('use of untyped nil in assignment (use `unsafe` | ${c.inside_unsafe})',
 								right.pos())
-						}
-						if mut right is ast.PrefixExpr && left_type.is_ptr()
-							&& right.right_type == ast.int_literal_type_idx {
-							if mut right.right is ast.Ident && right.right.obj is ast.ConstField {
-								c.error('cannot assign a pointer to a constant with an integer literal value',
-									right.right.pos)
-								c.add_error_detail('Specify the type for the constant value. Example:')
-								c.add_error_detail('         `const ${right.right.name.all_after_last('.')} = int(${(right.right.obj as ast.ConstField).expr})`')
-							}
 						}
 					}
 					mut ident_var_info := left.info as ast.IdentVar

--- a/vlib/v/checker/tests/assign_const_ptr_int_literal_err.out
+++ b/vlib/v/checker/tests/assign_const_ptr_int_literal_err.out
@@ -1,0 +1,14 @@
+vlib/v/checker/tests/assign_const_ptr_int_literal_err.vv:4:2: warning: unused variable: `b`
+    2 |
+    3 | fn main() {
+    4 |     b := &a
+      |     ^
+    5 | }
+Details: Specify the type for the constant value. Example:
+         `const a = int(3)`
+vlib/v/checker/tests/assign_const_ptr_int_literal_err.vv:4:8: error: cannot assign a pointer to a constant with an integer literal value
+    2 |
+    3 | fn main() {
+    4 |     b := &a
+      |           ^
+    5 | }

--- a/vlib/v/checker/tests/assign_const_ptr_int_literal_err.out
+++ b/vlib/v/checker/tests/assign_const_ptr_int_literal_err.out
@@ -1,14 +1,9 @@
-vlib/v/checker/tests/assign_const_ptr_int_literal_err.vv:4:2: warning: unused variable: `b`
-    2 |
-    3 | fn main() {
-    4 |     b := &a
-      |     ^
-    5 | }
-Details: Specify the type for the constant value. Example:
-         `const a = int(3)`
 vlib/v/checker/tests/assign_const_ptr_int_literal_err.vv:4:8: error: cannot assign a pointer to a constant with an integer literal value
-    2 |
+    2 | 
     3 | fn main() {
     4 |     b := &a
       |           ^
-    5 | }
+    5 |     dump(b)
+    6 | }
+Details: Specify the type for the constant value. Example:
+         `const a = int(3)`

--- a/vlib/v/checker/tests/assign_const_ptr_int_literal_err.vv
+++ b/vlib/v/checker/tests/assign_const_ptr_int_literal_err.vv
@@ -1,0 +1,5 @@
+const a = 3
+
+fn main() {
+	b := &a
+}

--- a/vlib/v/checker/tests/assign_const_ptr_int_literal_err.vv
+++ b/vlib/v/checker/tests/assign_const_ptr_int_literal_err.vv
@@ -2,4 +2,5 @@ const a = 3
 
 fn main() {
 	b := &a
+	dump(b)
 }

--- a/vlib/v/checker/tests/blank_ident_const_ptr_int_literal_err.out
+++ b/vlib/v/checker/tests/blank_ident_const_ptr_int_literal_err.out
@@ -4,3 +4,5 @@ vlib/v/checker/tests/blank_ident_const_ptr_int_literal_err.vv:4:8: error: cannot
     4 |     _ := &a
       |           ^
     5 | }
+Details: Specify the type for the constant value. Example:
+         `const a = int(3)`

--- a/vlib/v/checker/tests/blank_ident_const_ptr_int_literal_err.out
+++ b/vlib/v/checker/tests/blank_ident_const_ptr_int_literal_err.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/blank_ident_const_ptr_int_literal_err.vv:4:8: error: cannot assign a pointer to a constant with an integer literal value
+    2 | 
+    3 | fn main() {
+    4 |     _ := &a
+      |           ^
+    5 | }

--- a/vlib/v/checker/tests/blank_ident_const_ptr_int_literal_err.vv
+++ b/vlib/v/checker/tests/blank_ident_const_ptr_int_literal_err.vv
@@ -1,0 +1,5 @@
+const a = 3
+
+fn main() {
+	_ := &a
+}

